File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string str) {

        string temp="";
        vector<string>arr;

        for(int i=0;i<str.length();i++){

            if(str[i]==' ' || i==str.length()){

                if(temp.length()>0){
                    arr.push_back(temp);
                    temp="";
                }
            }

            else{

                temp+=str[i];
            }
        }

        if(temp.length()>0){
            arr.push_back(temp);
        }

        reverse(arr.begin(),arr.end());

        string ans="";

        for(int i=0;i<arr.size();i++){
            cout<<arr[i]<<" ";
            ans+=arr[i];
            ans+=' ';
        }

        ans.pop_back();
        return ans;
        
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(str) {
    let j=str.length-1;
    
    let temp="";
    let ans="";
    
    while(j>=-1){
        
        if(str[j]==' ' || j==-1 ){

        
        if(temp.length>=1){
            for(let i=temp.length-1;i>=0;i--){
                ans+=temp[i];
            }
            ans+=' ';
            temp="";
        }
        }
        else{
            temp+=str[j];
        }
        
        j--;
    }
    
    return ans.trim();
};